include/k4arecord/playback.hpp

```diff
@@ -18,7 +18,108 @@
 namespace k4a
 {
 
-/** \class playback playback.hpp
+/** \class data_block playback.hpp <k4arecord/playback.hpp>
+ * Wrapper for \ref k4a_playback_data_block_t
+ *
+ * \sa k4a_playback_data_block_t
+ */
+class data_block
+{
+public:
+    /** Creates a data_block from a k4a_playback_data_block_t
+     * Takes ownership of the handle, you should not call k4a_playback_data_block_release on the handle after
+     * giving it to the data_block; the data_block will take care of that.
+     */
+    data_block(k4a_playback_data_block_t handle = nullptr) noexcept : m_handle(handle) {}
+
+    // No Copies allowed
+    data_block(const data_block &) = delete;
+    data_block &operator=(const data_block &) = delete;
+
+    /** Moves another data_block into a new data_block
+     */
+    data_block(data_block &&other) noexcept : m_handle(other.m_handle)
+    {
+        reset();
+        m_handle = other.m_handle;
```

Comment:
You're doing this twice now, and I think potentially calling `k4a_playback_data_block_release` on the handle.

I think just removing ` : m_handle(other.m_handle)` would be fine.

Reply:
Calling `reset()` in the constructor shouldn't be required either, since it should always be a new object.